app/views/articles/edit.html.haml

```diff
@@ -0,0 +1,12 @@
+%h2 Edit Article
+
+= form_with model: @article , local: true do |form|
+  = form.label :headline
+  %br
+  = form.text_field :headline
+  %br
+  = form.label :content
+  %br
+  = form.text_field :content
+  %br
+  = form.submit 'Save Article'
```

Comment:
You have this same form on new.html.haml, extract the forms to a partial and render the partial on each page